### PR TITLE
[CIF-146] Set version constraints

### PR DESCRIPTION
**Pull request by jterry64**
## What changes were proposed in this PR?

After some research, it seems the best practice for dealing with dependency versions in libraries is:
- Versions in setup.py should be as lax as possible so we don't cause dependency conflicts with other libraries. Never set upper bounds, set lower bounds if we know there's a minimum version we need for it to work.
- For dev environment (conda in our case), pin specific versions so our environment is easily reproducible while developing

As far as I'm aware, there's no specific minimum version for our libraries, so didn't set any. But let me know if you can think of any. I also remove a couple dependencies that are no longer used in any imports.

## How was this patch tested?

Running tests and notebook with upgrade conda environment.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -3,24 +3,20 @@ channels:
...
+  - odc-stac=0.3.8
+  - pystac-client=0.7.5
+  - pytest=7.4.3
+  - xarray-spatial=0.3.7
+  - xee=0.0.3
+  - utm=0.7.0
+  - osmnx=1.8.1
+  - dask[complete]=2023.11.0
+  - matplotlib=3.8.2
+  - jupyter=1.0.0
```
Comment (chrowe) [inline environment.yml:19]: Do we want to use https://pypi.org/project/jupyterlab/ instead? Seems like this installs the latest version of Jupyter Lab so is not locking the version really, but maybe that is fine.

File: notebooks/tutorial/compute indicators.ipynb
Comment (chrowe) [inline notebooks/tutorial/compute indicators.ipynb:1]: I assume there are not relevant changes here.

**Review by chrowe — APPROVED** [review]

**Comment by chrowe** [comment]
@jterry64 I think this issue came up related to the old version of `xarray` that was producing warnings. Not sure if it is better  let people use an old version with warnings or require an newer version to eliminate the warnings.

**Comment by jterry64** [comment]
> @jterry64 I think this issue came up related to the old version of `xarray` that was producing warnings. Not sure if it is better let people use an old version with warnings or require an newer version to eliminate the warnings.

I think that's fine it it's just a deprecation warning, I don't think we want to force people to upgrade their xarray if they use our library unless there's clear errors. Using the conda environment among us should get rid of the warning.

**Review by weiqi-tori — APPROVED** [review]
Updated tutorial files to solve the conflicts

**Pull request merged**